articles/working-with-apis-in-typescript/index.md

```diff
@@ -0,0 +1,296 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/working-with-apis-in-typescript/
+title: Working with APIs in TypeScript
+description: In this article, we will take a look at how to harness the power of TypeScript to consume APIs.
+author: saiharsha-balasubramaniam
+date: 2021-04-14T00:00:00-17:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/working-with-apis-in-typescript/hero.png
+    alt: TypeScript API example image
+---
+
+TypeScript was introduced by Microsoft as a superset of JavaScript. It takes the versatility of JavaScript and improves it with type definitions. TypeScript is used with a lot of popular frameworks like React.js and React Native to build enterprise applications.
+
+<!--more-->
+
+In this article, we shall take a look at how to harness the power of TypeScript to create an API.
+
+### Prerequisites & Pre-Reading Material
+
+- A basic understanding of [HTTP Requests](https://www.section.io/engineering-education/http-requests-nodejs/) and [APIs](https://www.section.io/engineering-education/rest-api/).
```

Comment:
You need not type the entire URL if you are referencing a Section article. Just the path `/engineering-education/http-requests-nodejs/` would suffice.

Reply:
Understood :D